Sources/System/AsyncFileDescriptor.swift

```diff
@@ -0,0 +1,166 @@
+@_implementationOnly import CSystem
+
+public struct AsyncFileDescriptor: ~Copyable {
+    @usableFromInline var open: Bool = true
+    @usableFromInline let fileSlot: IORingFileSlot
```

Comment:
Ideally this should be a borrow of the file slot, but that's hard to express right now